topic-admin-operator/src/controller/mod.rs

```diff
@@ -0,0 +1,40 @@
+pub mod app;
+
+use serde::Deserialize;
+use std::{collections::HashMap, num::NonZeroU32};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct ControllerConfig {
+    #[serde(default = "default::num_partitions")]
+    pub num_partitions: NonZeroU32,
+    #[serde(default = "default::num_replicas")]
+    pub num_replicas: NonZeroU32,
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+    pub properties: HashMap<String, String>,
+}
```

Comment:
Maybe it should be possible to use the Kafka defaults for partitions and replicas? Kafka itself has 'global' defaults for these that will be used if not set.

Reply:
Maybe that is not supported by the admin client. It looks like I need to provide this information: https://docs.rs/rdkafka/0.27.0/rdkafka/admin/struct.NewTopic.html